lib/b-tree.js

```diff
@@ -0,0 +1,318 @@
+'use strict';
+
+const DEFAULT_DEGREE = 6; // min degree of b-tree.
+// All vertices except the root have [degree ... 2 * degree] child nodes
+
+function BTree(degree = DEFAULT_DEGREE) {
+  this.root = new BTreeNode();
+  this.minDegree = degree;
+}
+
+function BTreeNode(leaf = true) {
+  this.elements = [];
+  this.leaf = leaf;
+  this.lastChild = null;
+}
+
+function Element(value, typle) {
```

Comment:
You already have `leaf` property in `BTreeNode`, why not use it and store inherent data in elements?

Reply:
``leaf`` is just a flag that informs: is the current node a leaf or not? I dont quite understand how we can store data there?